Production_scripts/Social_Care/Care_Home.R

```diff
@@ -0,0 +1,416 @@
+  group_by(ch_postcode, ch_name) %>%
+  # Identify the closest match in case there are multiple close matches
+  mutate(
+    min_jaccard = min(match_distance_jaccard, na.rm = TRUE),
+    min_cosine = min(match_distance_cosine, na.rm = TRUE),
+    min_match_mean = min(match_mean, na.rm = TRUE)
+  ) %>%
+  ungroup()
+
+no_postcode_match <- matched_ch_data %>%
+  anti_join(ch_name_lookup, by = "ch_postcode")
+
+name_postcode_clean <- matched_ch_data %>%
+  # Remove records with no matching postcode, we'll add them back later
+  semi_join(ch_name_lookup, by = "ch_postcode") %>%
+  # Create a unique ID per row so we can get rid of duplicates later
+  mutate(ch_record_id = row_number()) %>%
+  # Match CH names with the generated metrics and the lookup. This will create
+  # duplicates which should be filtered out as we identify matches
+  left_join(ch_name_match_metrics, by = c("ch_postcode", "ch_name")) %>%
+  mutate(
+    # Work out the duration of the stay
+    # If the end date is missing set this to the end of the quarter
+    stay_interval = interval(
+      ch_admission_date,
+      min(ch_discharge_date, record_date, na.rm = TRUE)
+    ),
+    # Highlight which stays overlap with an open care home name
+    stay_overlaps_open = int_overlaps(stay_interval, open_interval) & (int_start(stay_interval) >= int_start(open_interval)),
+    # Hightlight which names seem to be good matches
+    name_match = case_when(
+      # Exact match
+      ch_name == ch_name_validated ~ TRUE,
+      # Submitted name is missing and stay dates are valid for the CH
+      is.na(ch_name) & stay_overlaps_open ~ TRUE,
+      # This name had the closest 'jaccard' distance of all possibilities
+      (min_jaccard == match_distance_jaccard) & match_distance_jaccard <= 0.25 ~ TRUE,
+      # This name had the closest 'cosine' distance of all possibilities
+      (min_cosine == match_distance_cosine) & match_distance_cosine <= 0.3 ~ TRUE,
+      # This name had the closest 'mean' distance (used when the above disagree)
+      (min_match_mean == match_mean) & match_mean <= 0.25 ~ TRUE,
+      # No good match
+      TRUE ~ FALSE
+    )
+  ) %>%
+  # Group by record (There will be duplicate rows per record if there multiple 'options' for the possible CH name)
+  group_by(ch_record_id) %>%
+  mutate(
+    # Highlight where the record has no matches out of any of the options
+    no_name_matches = all(!name_match),
+    # Highlight where the record has no overlaps (in dates) with any of the options
+    no_overlaps = all(!stay_overlaps_open)
+  ) %>%
+  # Keep a record if:
+  # 1) It's name matches `name_match`
+  # Or either 2)a) None of the option's names match AND this option overlaps in dates (e.g. the submitted name is missing but the dates match)
+  # or 2)b) None of the option's names match AND none of the dates overlap (i.e. we don't have any idea what name to use)
+  filter(n() == 1 | sum(name_match) == 1 | all(!name_match)) %>%
```

Comment:
```suggestion
  filter(n() == 1 | sum(name_match) == 1 | all(name_match)) %>%
```
I think it's this line which is causing the issue. Is `all(!name_match)` picking up names which are false here?